cli/src/commands/log.rs

```diff
@@ -49,7 +49,7 @@ pub(crate) struct LogArgs {
     /// Limit number of revisions to show
     ///
     /// Applied after revisions are filtered and reordered.
-    #[arg(long, short)]
+    #[arg(long, short = 'n', visible_short_alias = 'l')]
```

Comment:
Could you also add a mention in changelog that `-l` is now deprecated?

Reply:
If we are *deprecating* `l`, it might make sense to use `short_alias` here instead of `visible_short_alias`.

Currently, `jj help log` looks like this:

```
...
  -n, --limit <LIMIT>
          Limit number of revisions to show

          Applied after revisions are filtered and
          reordered.

          [short aliases: l]
...
```

(This doesn't affect the generated docs due to limitations of `clamp_markdown`. Unfortunately, you have to try `jj help log` to know).

Reply:
I think we don't *have* to have a deprecation warning, by the way, hiding the alias gets us most of the way there. Adding `-l` as a separate short option just to print the warning, as you suggested in the description, also seems fine.

Reply:
I went with a separate hidden argument which warns on use and amended the Changelog.md